ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/df_driver.py

```diff
@@ -118,21 +130,38 @@ def parse_and_process_data_preprocessing_config(object_storage_client, spark, ge
                             if 'distinct_categories' not in metadata:
                                 metadata["distinct_categories"] = dict()
                             metadata["distinct_categories"][step["args"]["category"]] = distinct_categories
-                        elif phase == "inference":
+                        elif phase == INFERENCING:
                             step["args"]["distinct_categories"] = metadata["distinct_categories"][step["args"]["category"]]
                             _, dfs[step_config["configurations"]["dataframeName"]] = \
-                                eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])
+                                eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])                            
                     # sharding also needs to be specially treated. We need:
                     # 1. Check whether it's the last one of the processing_steps
                     # 2. Saving multiple dataframes
-                    # TODO: Check in sharding program to make sure user
                     elif func_name == "sharding":
                         step_config_len = len(step_config["configurations"]["steps"]) - 1
                         if (step_config["configurations"]["steps"]).index(step) != step_config_len:
                             raise Exception("Sharding must be the last of processing steps")
                         sharding_dfs = eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])
                         sharding_dict = list(sharding_dfs.keys())
                         dfs.update(sharding_dfs)
+                    # Pivoting also needs specific treatment since it's also a data dependent processing
+                    # Usually pivoting will be conducted after merge and joining if multiple datasets involves
+                    # If training, we should pick out the distinct item of assigned column and store it inside metadata
+                    # If inferencing, we should read pre-stored distinct_column_values from metadata pass it for pivoting function
+                    elif func_name == "spark_pivoting":
+                        if phase == TRAINING:
+                            distinct_column_values, dfs[step_config["configurations"]["dataframeName"]] = \
+                                eval(func_name)(dfs[step_config["configurations"]["dataframeName"]], **step["args"])
+                            flat_distinct_column_values = [item for sublist in distinct_column_values for item in sublist]
+                            # if 'distinct_column_values' not in metadata:
```

Comment:
nit: clean up commented code that will not be used in future

Reply:
nice catch! cleaned up